applications/StructuralMechanicsApplication/custom_conditions/base_load_condition.cpp

```diff
@@ -19,330 +20,401 @@
+/***********************************************************************************/
 
-        for (unsigned int i = 0; i < NumberOfNodes; i++)
-        {
-            const array_1d<double, 3 > & Displacement = GetGeometry()[i].FastGetSolutionStepValue(DISPLACEMENT, Step);
-            unsigned int index = i * dim;
-            for(unsigned int k = 0; k < dim; ++k)
-            {
-                rValues[index + k] = Displacement[k];
-            }
-        }
+void BaseLoadCondition::FinalizeNonLinearIteration( ProcessInfo& rCurrentProcessInfo )
+{
+    // TODO: Add somethig if necessary
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void BaseLoadCondition::FinalizeSolutionStep( ProcessInfo& rCurrentProcessInfo )
+{
+    // TODO: Add somethig if necessary
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void BaseLoadCondition::EquationIdVector(
+    EquationIdVectorType& rResult,
+    ProcessInfo& rCurrentProcessInfo
+    )
+{
+    KRATOS_TRY
+
+    const SizeType number_of_nodes = GetGeometry().size();
+    const SizeType dim = GetGeometry().WorkingSpaceDimension();
+    const SizeType block_size = this->GetBlockSize();
+    if (rResult.size() != dim * number_of_nodes) {
+        rResult.resize(number_of_nodes * block_size, false);
     }
 
-    //***********************************************************************
-    //***********************************************************************
-
-    void BaseLoadCondition::GetFirstDerivativesVector(
-        Vector& rValues,
-        int Step
-        )
-    {
-        const unsigned int NumberOfNodes = GetGeometry().size();
-        const unsigned int dim = GetGeometry().WorkingSpaceDimension();
-        const unsigned int MatSize = NumberOfNodes * dim;
-
-        if (rValues.size() != MatSize)
-        {
-            rValues.resize(MatSize, false);
-        }
+    const SizeType pos = this->GetGeometry()[0].GetDofPosition(DISPLACEMENT_X);
 
-        for (unsigned int i = 0; i < NumberOfNodes; i++)
-        {
-            const array_1d<double, 3 > & Velocity = GetGeometry()[i].FastGetSolutionStepValue(VELOCITY, Step);
-            const unsigned int index = i * dim;
-            for(unsigned int k = 0; k<dim; ++k)
-            {
-                rValues[index + k] = Velocity[k];
-            }
+    if(dim == 2) {
+        for (SizeType i = 0; i < number_of_nodes; ++i) {
+            const SizeType index = i * block_size;
+            rResult[index    ] = GetGeometry()[i].GetDof(DISPLACEMENT_X,pos    ).EquationId();
+            rResult[index + 1] = GetGeometry()[i].GetDof(DISPLACEMENT_Y,pos + 1).EquationId();
+            if (this->HasRotDof())
```

Comment:
I don't fully understand this check:
`virtual bool HasRotDof(){return (GetGeometry()[0].HasDofFor(ROTATION_X) && GetGeometry().size() == 2);};`

Reply:
It was already there...